app/controllers/toezicht/inzendingen.js

```diff
@@ -18,22 +23,47 @@ export default Controller.extend({
     this.set('header', ENV['vo-webuniversum']['header']);
   },
 
-  filterChanged: observer('bestuurseenheidId', 'classificatieId', 'provincieId', 'besluitTypeId',
-                          'sessionDateFrom', 'sessionDateTo', 'sentDateFrom', 'sentDateTo', 'statusId', function() {
+  filterChanged: observer('bestuurseenheidId', 'classificatieId', 'provincieId',
+                          'besluitType', 'toezichtRegulationTypeId', 'sessionDateFrom', 'sessionDateTo', 'sentDateFrom', 'sentDateTo', 'statusUri', function() {
     this.set('page', 0);
   }),
 
+  isRegulation: computed('besluitType.isRegulation', function(){
+    if (this.besluitType === undefined && this.besluitTypeId !== undefined)
+      this.set('besluitType', this.store.findRecord('besluitType', this.besluitTypeId));
+    return this.get('besluitType.isRegulation');
+  }),
+
+  besluitTypeUri: computed('besluitType.uri', function(){
+    return this.get('besluitType.uri');
+  }),
+
   actions: {
+    setToTreatStatus(event) {
+      this.set('statusUri', null);
+      if(event.target.checked) {
+        this.set('statusUri', this._toTreatStatusUri);
+      }
+    },
+
     resetFilters() {
       ['bestuurseenheidId',
        'classificatieId',
        'provincieId',
        'besluitTypeId',
+       'toezichtRegulationTypeId',
        'sessionDateFrom',
        'sessionDateTo',
        'sentDateFrom',
        'sentDateTo',
-       'statusId'].forEach(filter => this.set(filter, null));
+       'statusUri'].forEach(filter => this.set(filter, null));
+    },
+
+    readBesluitType(type) {
+      this.set('besluitType', type);
+      this.set('besluitTypeId', type ? type.id : null);
+      if (!type)
+        this.set('regulationTypeId', null);
```

Comment:
We want to reset the `regulationTypeId` in all cases independent whether the newly selected type is `null`.

Reply:
Silly mistake!! the if statement is already taken care of in line 64. The if statement is redundant. This is because I had first removed line 64 then later added it and forgotten to remove lines 65-66

Reply:
Resetting `regulationTypeId` to `null` is correct, but it needs to be done in both cases, when `type` is `null` but also when `type` is not `null`.